Created Navigation Bar

roundband5 opened the pull request:
Nav bar includes 2 navigation buttons (Home, About), Website logo in the center, and Search bar on the right
Desktop view (1920x1080)
![image](https://user-images.githubusercontent.com/30790329/126413253-f79873b7-4338-4400-a26e-b57fcebeb66e.png)

Smaller view (1134x686 view)
![image](https://user-images.githubusercontent.com/30790329/126413310-f6bf8411-8488-47f8-8c82-4b6b361cacdf.png)

lchua2314 commented on src/modules/Navbar/Navbar.css:
```diff
@@ -0,0 +1,89 @@
+.NavbarItems {
+    background: linear-gradient(90deg, rgb(0, 0, 0) 0%, rgb(6, 6, 7) 100%);
+    height: 100px;
```
Not sure about hardcoded lengths.

lchua2314 replied:
Not saying that you should change it, but in Android you want to use scaling units since hardcoded lengths do not scale on different sized screens.

lchua2314 replied:
https://gist.github.com/basham/2175a16ab7c60ce8e001

lchua2314 commented on src/modules/Navbar/Navbar.css:
```diff
@@ -0,0 +1,89 @@
+    color:white;
+    position:absolute;
+    left: 50%;
+    margin-left: -50px !important;  /* 50% of your logo width */
```
!important is used to override all the other CSS "margin-left"s. I think you just want to leave the comment.

lchua2314 replied:
Hardcoded pixels as well.

lchua2314 replied:
Also, I would not recommend using !important for the future because I think it's not best practice.

lchua2314 commented on src/modules/Navbar/MenuItems.js:
```diff
@@ -0,0 +1,12 @@
+export const MenuItems= [
+    {
```
Please use 2 space indent for everything.

lchua2314 commented on src/modules/Navbar/Navbar.js:
```diff
@@ -0,0 +1,50 @@
+                </ul>
+                <h1 className="navbar-logo">
+                    <img src={AppLogo} className="App-logo" alt="Harbinger Musik Logo" width="90" height="90"
+                    />
```
Keep this in the line above.

lchua2314 replied:
Hardcoded logo size.

lchua2314 commented on src/modules/Navbar/Navbar.js:
```diff
@@ -0,0 +1,50 @@
+                    </form>
+                </div>
+            </nav>
+            
```
Remove random empty line.

lchua2314 commented on src/modules/Navbar/Navbar.js:
```diff
@@ -0,0 +1,50 @@
+    render(){
+        return(
+            <nav className="NavbarItems">
+                
```
Please remove empty lines unless they are to help with readability.

lchua2314 commented on src/modules/Navbar/Navbar.js:
```diff
@@ -0,0 +1,50 @@
+                        <button type="submit"><img src={SearchIcon} className="Search-Icon" width="40" height="40"
+                    /><i class="fa fa-search"></i></button>
+                    </form>
+                </div>
```
I was thinking of just using an import for the search bar since I think it can easily handle the functionality. Just keep this for now.

lchua2314 reviewed (changes requested):
Fix the code where I left comments (also try to fix all the hardcoded sizes). Other than that, the navbar looks good.

lchua2314 commented on src/modules/Navbar/Navbar.css:
```diff
@@ -0,0 +1,89 @@
+
+
+.navbar-logo{
+    color:white;
```
Try not to use pure white and black colors. This can be bad for people's eyes. Try lowering the white color to #EEEEEE instead of using the default white (#FFFFFF).

lchua2314 commented on src/modules/Navbar/Navbar.css:
```diff
@@ -0,0 +1,89 @@
+}
+
+.nav-links{
+    color:white;
```
Lower pure white color.

lchua2314 commented on src/modules/Navbar/Navbar.css:
```diff
@@ -0,0 +1,89 @@
+}
+
+.fa-bars{
+    color:white
```
Lower pure white color

leonzh2k reviewed (commented):
How were you planning to use PHP for the form?

roundband5 commented:
Fixed formatting issues and added Router link. About and Home page links now working.

![ezgif-3-1b5ace73102f](https://user-images.githubusercontent.com/30790329/126711146-a83eda61-491d-4d73-a3e6-eca7bc6ddab4.gif)

lchua2314 commented on src/App.js:
```diff
@@ -1,9 +1,22 @@
+import './App.css';
+import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
+import Home from './modules/Home/Home';
+import About from './modules/About/About'
```
Missing semicolon

lchua2314 commented on src/App.js:
```diff
@@ -1,9 +1,22 @@
+        <Route path='/About' component={About} />
+      </Switch>
+    </Router>
+
```
Remove empty line

lchua2314 commented on src/modules/Navbar/Navbar.css:
```diff
@@ -0,0 +1,87 @@
+  text-align:right;
+  font-weight:bold;
+  padding:2px;
+}
```
Add an empty line here. Same mistake I did.

lchua2314 commented on src/modules/Navbar/Navbar.css:
```diff
@@ -0,0 +1,87 @@
+}
+
+.nav-links:hover {
+  background-color:violet;
```
Violet doesn't fit with the theme, but we don't really care about the theme right now so that doesn't need to be changed atm.

lchua2314 commented on src/modules/Navbar/Navbar.css:
```diff
@@ -0,0 +1,87 @@
+  color:#EEEEEE
+}
+
+/*coming soon*/
```
poggers.
Add two spaces so it looks like this:
/* coming soon */

lchua2314 commented on src/modules/Navbar/Navbar.js:
```diff
@@ -0,0 +1,47 @@
+          </i>
+        </h1>
+
+        <div class="search-container">
```
Change "class" to "className". It's a convention to avoid confusion between "class" keyword in JS and JSX.

lchua2314 commented on src/modules/Navbar/Navbar.js:
```diff
@@ -0,0 +1,47 @@
+          <form action="/action_page.php">
+            <input type="text" placeholder="" name="search"></input>
+            <button type="submit"><img src={SearchIcon} className="Search-Icon" width="40" height="40" />
+              <i class="fa fa-search"></i>
```
Change this to className as well.

lchua2314 reviewed (changes requested):
Change a few things bit. Everything else looks solid.
